tls/extensions/s2n_client_supported_groups.c

```diff
@@ -19,15 +19,19 @@
 #include "tls/extensions/s2n_client_supported_groups.h"
 #include "tls/s2n_tls.h"
 #include "tls/s2n_tls_parameters.h"
-#include "tls/s2n_ecc_preferences.h"
+#include "tls/s2n_security_policies.h"
 
 #include "utils/s2n_safety.h"
 
 int s2n_extensions_client_supported_groups_send(struct s2n_connection *conn, struct s2n_stuffer *out)
 {
     notnull_check(conn);
-    const struct s2n_ecc_preferences *ecc_pref = conn->config->ecc_preferences;
-    notnull_check(ecc_pref);
+
+    const struct s2n_security_policy *security_policy = NULL;
```

Comment:
these 5 lines of code seems to be used everywhere - it not only becomes a little unwieldy but leaves room for error if they're copied without its entirety, perhaps think of a way to replace the block with a macro instead?

Reply:
Agreed, I'll look into a macro for it.